### DISCUSSION TITLE
`whenNotNull` and `whenHas` support

### DISCUSSION THREAD

**Discussion by alex-watchenterprise** [dedoc/scramble#622] [Ideas]
`v0.11.25`

Scramble doing a great job with recognizing as optional

- merge
- mergeWhen
- when 

However, it will be very good to also add a support for 

- whenHas
- whenNotNull

because now we need to do something like this:

`$this->when($this->title, $this->title)`

**Comment by romalytvynenko** [comment]
@alex-watchenterprise released in `v0.11.30`
